lib/llvm-backend/build.rs

```diff
@@ -262,6 +262,46 @@ fn is_llvm_debug() -> bool {
     llvm_config("--build-mode").contains("Debug")
 }
 
+fn get_llvm_target_name() -> String {
+    let name = if cfg!(all(target_os = "macos", target_arch = "x86_64")) {
+        "x86_64-apple-darwin"
+    } else if cfg!(all(target_os = "linux", target_arch = "x86_64")) {
+        "x86_64-linux-gnu-ubuntu-16.04"
+    } else {
+        panic!("Unsupport host for installing llvm")
+    };
+
+    format!("clang+llvm-8.0.0-{}", name)
+}
+
+fn find_llvm_url() -> String {
+    let name = get_llvm_target_name();
+    format!("https://releases.llvm.org/8.0.0/{}.tar.xz", name)
+}
+
+fn install_llvm() {
+    let llvm_path: PathBuf = format!("{}/llvm", std::env::var("OUT_DIR").unwrap()).into();
+
+    std::env::set_var(
+        &*ENV_LLVM_PREFIX,
+        format!("{}/{}", llvm_path.display(), get_llvm_target_name()),
+    );
+
+    if llvm_path.exists() {
+        return;
+    }
+
+    let url = find_llvm_url();
+
+    let resp = reqwest::get(&url).expect("Failed to connect to the llvm server");
+    let resp = lzma::LzmaReader::new_decompressor(resp).expect("Failed to initialize decompressor");
```

Comment:
Please have a list of expected sha256sums and verify that the download matches before doing anything else with the file.